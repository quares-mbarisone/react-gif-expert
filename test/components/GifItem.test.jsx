import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en Componente <GifItem />', () => {

    const title = "Luffy Gomu Gomu no Bazuka"
    const url= "https://one-piece.com/luffy.jpg"

    test('debe de hacer match con la snapshot', () => {
        const { container } = render( <GifItem title={title} url={url} /> )
        expect( container ).toMatchSnapshot();
    })

    test('debe mostrarme la imagen con el URL y el Alt indicado', () => {
        render( <GifItem title={title} url={url} /> );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })

    test('debe mostrarme el titulo en el componente', () => {
        render( <GifItem title={title} url={url} /> );
        expect( screen.getByText( title ) ).toBeTruthy();
    })
})