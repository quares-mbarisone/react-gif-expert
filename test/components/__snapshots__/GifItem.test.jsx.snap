// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Componente <GifItem /> debe de hacer match con la snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Luffy Gomu Gomu no Bazuka"
      src="https://one-piece.com/luffy.jpg"
    />
    <p>
       
      Luffy Gomu Gomu no Bazuka
       
    </p>
  </div>
</div>
`;
